gen/sam/airdefensegroupgenerator.py

```diff
@@ -46,31 +30,17 @@ class AirDefenseGroupGenerator(VehicleGroupGenerator[SamGroundObject], ABC):
     def __init__(self, game: Game, ground_object: SamGroundObject) -> None:
         super().__init__(game, ground_object)
 
-        self.vg.name = self.group_name_for_role(self.vg.id, self.primary_group_role())
+        self.iads_roles: List[IADSRole] = [self.range().default_role]
         self.auxiliary_groups: List[VehicleGroup] = []
         self.heading = self.heading_to_conflict()
 
-    def add_auxiliary_group(self, role: SkynetRole) -> VehicleGroup:
+    def add_auxiliary_group(self, role: IADSRole) -> VehicleGroup:
         gid = self.game.next_group_id()
-        group = VehicleGroup(gid, self.group_name_for_role(gid, role))
+        group = VehicleGroup(gid, f"{self.go.group_name}|{role.value}")
         self.auxiliary_groups.append(group)
+        self.iads_roles.append(role)
         return group
 
-    def group_name_for_role(self, gid: int, role: SkynetRole) -> str:
-        if role is SkynetRole.NoSkynetBehavior:
-            # No special naming needed for air defense groups that don't participate in
-            # Skynet.
-            return f"{self.go.group_name}|{gid}"
-
-        # For those that do, we need a prefix of `$COLOR|SAM| so our Skynet config picks
-        # the group up at all. To support PDs we need to append the ID of the TGO so
-        # that the PD will know which group it's protecting. We then append the role so
-        # our config knows what to do with the group, and finally the GID of *this*
-        # group to ensure no conflicts.
-        return "|".join(
-            [self.go.faction_color, "SAM", str(self.go.group_id), role.value, str(gid)]
-        )
```

Comment:
As the special naming was removed maybe other plugins which depend on the names could have problems. JTAC is fine, i checked it but i suspect LotATC to have problems.